cmd/attest_test.go

```diff
@@ -0,0 +1,199 @@
+package cmd
+
+import (
+	"os"
+	"testing"
+
+	"github.com/google/go-tpm-tools/client"
+	"github.com/google/go-tpm-tools/internal/test"
+)
+
+func makeOutputFile(tb testing.TB, output string) string {
+	tb.Helper()
+	file, err := os.CreateTemp("", output)
+	if err != nil {
+		tb.Fatal(err)
+	}
+	defer file.Close()
+	return file.Name()
+}
+
+func TestNonce(t * testing.T){
+	rwc := test.GetTPM(t)
+	defer client.CheckedClose(t, rwc)
+	ExternalTPM = rwc
+
+	//without nonce
+	RootCmd.SetArgs([]string{"attest","--key","AK"})
+	if err := RootCmd.Execute(); err == nil {
+		t.Error(err)
+	}
+
+	//with odd length nonce
+	RootCmd.SetArgs([]string{"attest","--nonce","12345","--key","AK"})
```

Comment:
gofmt should be adding in the spaces between your commas. Please lint your code.

Reply:
Thankyou , I will do gofmt before updating the pull request.

Reply:
I have used gofmt in the new commits . Please review it once.
For now I am resolving this comment.